pkg/rebuild/rebuild/rebuildremote.go

```diff
@@ -441,10 +441,15 @@ func doCloudBuild(ctx context.Context, client gcb.Client, build *cloudbuild.Buil
 	bi.BuildID = build.Id
 	bi.Steps = build.Steps
 	bi.BuildImages = make(map[string]string)
+	buildErr := gcb.ToError(build)
+	// Only return errors about build.Results if the build was otherwise error-free.
+	if buildErr == nil && (build.Results == nil || build.Results.BuildStepImages == nil || len(build.Results.BuildStepImages) != len(bi.Steps)) {
```

Comment:
I think this is a bug in ToError. Why is it not emitting an error in this case?

Reply:
It wasn't getting to gcb.ToError because it was panic-ing on a nil reference deref

Reply:
Ah wait I didn't see that ToError was called after the loop in the current code.

Yeah I think just lazily populating the field with the data available is fine. I don't even think additional checks are necessary since len(BuildStepImages) == len(Steps) is a documented behavior of the API in the case of a successful build. As such, the attestation code could assume it's populated.

Reply:
How's that?